devtools/platform-descriptor-json/src/test/resources/fakeextensions.json

```diff
@@ -1,5 +1,27 @@
 
 {
+    "any-future-property" : "blah",
+    "metadata":{
+        "application-properties": {
+            "quarkus.native.builder-image" : "foo-bar"
+        },
+        "maven" :  {
+            "repositories" : {
+                "id": "redhat",
+                "url": "https://maven.repository.redhat.com"
```

Comment:
@gastaldi @aloubyansky mind that when we actually set this data, we will need:

```json
"repositories" : {
    "id": "redhat",
    "url": "https://maven.repository.redhat.com",
    "releases-enabled": "true",
    "snapshots-enabled": "false"
},
"plugin-repositories" : {
    "id": "redhat",
    "url": "https://maven.repository.redhat.com",
    "releases-enabled": "true",
    "snapshots-enabled": "false"
}

```

Reply:
@gastaldi @aloubyansky  well it's also an array:
```json
            "repositories" : [
                {
                    "id": "redhat",
                    "url": "https://maven.repository.redhat.com",
                    "releases-enabled": "true",
                    "snapshots-enabled": "false"
                }
            ],
            "plugin-repositories" : [
                {
                    "id": "redhat",
                    "url": "https://maven.repository.redhat.com",
                    "releases-enabled": "true",
                    "snapshots-enabled": "false"
                }
            ]
```

Reply:
@ia3andy that's not a problem, the changes in this PR only expose the metadata attribute as a Map that can contain other Maps. Also @maxandersen suggested using a separate `snapshot-repositories` instead of the `snapshot-enabled` flag which I think it makes sense :)

Reply:
@gastaldi nop, it's a bad idea (sorry @maxandersen :)) it would be really hard to consume in the templates if we used `snapshot-repositories`.. let's stick to the maven structure..